api/kv.go

```diff
@@ -11,13 +11,33 @@ import (
 
 // KVPair is used to represent a single K/V entry
 type KVPair struct {
-	Key         string
+	// Key is the name of the key. It is also part of the URL path when accessed
+	// via the API.
+	Key string
+
+	// CreateIndex holds the index corresponding the creation of this KVPair. This
+	// is a read-only field.
 	CreateIndex uint64
+
+	// ModifyIndex is used for the Check-And-Set operations.
```

Comment:
I'd change this to:

ModifyIndex is used for the Check-And-Set operations and can also be fed into the WaitIndex of the QueryOptions in order to perform blocking queries.